dlt/common/libs/ibis.py

```diff
@@ -0,0 +1,233 @@
+    metadata to register what table exists and create bound expressions.
+
+    For example, if you use Ibis with the Snowflake backend, you will get use Ibis'
+    implementation. If you use the DltBackend with a Snowflake destination, you will
+    use the dlt SQL client for Snowflake. The SQL query created should be the same
+    because dlt internally uses the `ibis -> sqlglot` compiler, but the actual
+    execution might differ. This is especially likely when calling `.execute()` or
+    other methods that return data in memory.
+    """
+
+    name: str = "dlt"
+    supports_temporary_tables = False
+    supports_python_udfs = False
+
+    @classmethod
+    def from_dataset(cls, dataset: dlt.Dataset) -> _DltBackend:
+        """Create an Ibis `DltBackend` from a `dlt.Dataset`
+
+        This enables `dlt.Relation.to_ibis()` to create bound tables.
+        """
+        # NOTE it's essential to keep this lazy, no round-trip to destination
+        # sync with destination should be made through the dataset.
+        new_backend = cls()
+        new_backend._dataset = dataset
+        new_backend.compiler = _get_ibis_to_sqlglot_compiler(new_backend._dataset._destination)  # type: ignore[arg-type]
+        return new_backend
+
+    def to_native_ibis(self, *, read_only: bool = False) -> BaseBackend:
+        return get_native_ibis_backend(self._dataset, read_only=read_only)
+
+    @property
+    # @override
+    def version(self) -> str:
+        import importlib.metadata
+
+        return importlib.metadata.version("dlt")
+
+    # NOTE can change signature
+    # @override
+    def do_connect(
+        self,
+        destination: TDestinationReferenceArg,
+        dataset_name: str,
+        schema: Union[DltSchema, str, None] = None,
+        **config: Any,
+    ) -> None:
+        self._dataset = dlt.dataset(
+            destination=destination,
+            dataset_name=dataset_name,
+            schema=schema,
+        )
+        self.compiler = _get_ibis_to_sqlglot_compiler(self._dataset._destination)  # type: ignore[arg-type]
+
+    # derived from Ibis Snowflake implementation
+    @contextlib.contextmanager
+    # @override
+    def _safe_raw_sql(self, query: sge.ExpOrStr, **kwargs: Any) -> Any:
+        with contextlib.suppress(AttributeError):
+            query = _transpile(query, target_dialect=self.compiler.dialect)
+
+        with contextlib.closing(self.raw_sql(query, **kwargs)) as cur:
+            yield cur
+
+    # derived from Ibis Snowflake implementation
+    # @override
+    def raw_sql(self, query: Union[str, sg.Expression], **kwargs: Any) -> Any:
+        """Execute SQL string or SQLGlot expression using the dlt destination SQL client"""
+        with contextlib.suppress(AttributeError):
+            query = _transpile(query, target_dialect=self.compiler.dialect)
+
+        assert isinstance(query, str)
+        with self._dataset.sql_client as client:
```

Comment:
why you cannot call `dataset.query` instead. this will also transpile the query for you (IMO).